app/code/community/Newsletter2go/Apiextension/Model/Api2/Subscriber/Rest/Admin/V1.php

```diff
@@ -234,25 +248,46 @@ private function getSubscribersOnly($subscribed, $limit, $offset, $fields, $emai
     /**
      * This method replaces option id's with actual values for select and multi select field input types.
      *
-     * @param Values of fields for a single Customer $customer
-     * @return Values of fields for a single Customer with values of custom fields
+     * @param array $customer Values of fields for a single Customer $customer
+     *
+     * @return array Values of fields for a single Customer with values of custom fields
      */
-    private function getSelectValues ($customer){
-
-        foreach ($customer as $key => $value){
-            $read = Mage::getSingleton('core/resource')->getConnection('core_read');
-            $query = "SELECT * FROM eav_attribute WHERE attribute_code='". $key ."' AND frontend_input IN ('multiselect','select') AND is_user_defined = 1;";
-            $row = $read->fetchAll($query);
-            if (isset($row) && !empty($row)) {
-                $query = "SELECT value FROM eav_attribute_option_value WHERE option_id IN (" . $value . ");";
-                $values = $read->fetchAll($query);
-                $customer[$key] = null;
-                foreach ($values as $data) {
-                    $customer[$key][] =  $data['value'];
-                }
-                $customer[$key] = implode(", ", $customer[$key]);
+    private function getSelectValues($customer)
+    {
+        foreach ($customer as $key => $value) {
+            if (!array_key_exists($key, $this->attributeCache)) {
+                continue;
+            }
+
+            if (is_array($value)) {
+                $customer[$key] = implode(', ', array_values($this->attributeCache[$key]));
+            } else {
+                $customer[$key] = $this->attributeCache[$key][$value];
             }
         }
+
         return $customer;
     }
+
+    /**
+     * Builds attribute cache with value label pairs
+     */
+    private function buildAttributesCache()
+    {
+        /** @var Mage_Customer_Model_Resource_Attribute_Collection $attributes */
+        $attributes = Mage::getResourceModel('customer/attribute_collection');
+        $attributes->addFieldToFilter('frontend_input', array('in' => array('multiselect', 'select')));
+        $attributes->addFieldToFilter('is_user_defined', 1);
+        $items = $attributes->getItems();
+
+        /** @var Mage_Customer_Model_Attribute $attribute */
+        foreach ($items as $attribute) {
+            $attributeCode = $attribute->getAttributecode();
```

Comment:
is getAttributecode really without camel-case?

Reply:
Apparently, it is not: https://github.com/OpenMage/magento-mirror/search?utf8=%E2%9C%93&q=getAttributecode+&type=
Good catch.